test/api/publish.test.js

```diff
@@ -21,21 +20,27 @@ import {
 beforeEach(() => MockStorage.cleanUp());
 afterEach(() => MockStorage.cleanUp());
 
-function verifyAndDecodeJwtToken(res) {
+function verifyAndDecodeJwtToken(res): string {
+  if (!res.text) {
+    throw new Error(
+      `There was no 'text' property in the response, which shouldn't happen.`
+    );
+  }
+
   const token = res.text;
   const secret = config.jwtSecret;
   const decodedPayload = jwt.verify(token, secret, { algorithms: ['HS256'] });
-  res.text = decodedPayload.profileToken;
```

Comment:
Previously we were replacing the `text` (which is the response's data) with the token so that we could use supertest's `expect` thingie. Now it's a much more classic function :-)